transcom/mymove
# [MB-16348] backend adding date of contact for service items

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16348)

## Summary

The purpose of this PR is to include a date of attempted contact that is to be displayed in conjunction with `timeMilitary` for the  `Customer Contact` field. So instead of reading `Customer contact 1: 0400Z` (the current state) we are expecting it to read `Customer contact 1: 11 Nov 2022, 0400Z`.

This PR includes a migration for the new column along with associated updates to yaml files, models, payloads, prime ui, etc.

The changes are pretty straightforward and I think I covered everything so there's nothing in particular that im looking for extra scrutiny on but any input is greatly appreciated on this!

Check out [this slack thread if you would like extra context](https://ustcdp3.slack.com/archives/CTNTFJSBA/p1686772959097629)

Front end portion of this can be found [here](https://github.com/transcom/mymove/pull/10932) (My PR should probably be merged first)

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Run `make server_test`
2. All tests should pass
3. Create a new customer `HHG` move and move it along the system to make it available to the prime
4. Login into to the Prime UI and access your newly created move
5. Click the `Add Service Item` button
6. Scroll down to the form and select `Destination SIT` from the `Service item type` dropdown
7. The `date of attempted contact` field should be present - select a date, complete the form, and submit
8. It should now be reflected in the `mto_service_item_customer_contacts` table if you check your DB
9. You can also access the move as a `TOO` and check the query explorer to verify 🎉 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
Query Explorer:
![image](https://github.com/transcom/mymove/assets/111928238/bf8aac95-4511-4440-91a5-13be1fd440b8)

`mto_service_item_customer_contacts` Table:
![image](https://github.com/transcom/mymove/assets/111928238/90cf11bf-7a32-447e-889f-df9db0669c28)

Prime UI:
![image](https://github.com/transcom/mymove/assets/111928238/a1bb64b9-6dab-46a0-9a7e-8a1e5ca0af13)
